app/controllers/sessions_controller.rb

```diff
@@ -1,15 +1,20 @@
 class SessionsController < ApplicationController
-  after_filter "get_login_redirect", only: [:new]
+  after_action "get_login_redirect", only: [:new]
 
   def new
-
   end
 
   def create
     @user = User.find_by(username: params[:session][:username])
     if @user && @user.authenticate(params[:session][:password])
       session[:user_id] = @user.id
-      redirect_to dashboard_path
+      
+      if session[:login_redirect].include?("items")
+         redirect_to session[:login_redirect]
```

Comment:
This looks like a good way to handle the redirect. Makes it very easy to know where they're coming from. Does this save the params as well as the path? I.e. it has to save the slug if it's redirecting to a path that requires a slug, but not if it's just redirecting to something like the main index.